Framework/MDAlgorithms/src/CompareMDWorkspaces.cpp

```diff
@@ -31,6 +32,153 @@ class CompareFailsException : public std::runtime_error {
 namespace Mantid {
 namespace MDAlgorithms {
 
+class SimpleMDEvent {
+
+private:
+  std::vector<float> mCoordinates;
+  float mSignal;
+  float mError;
+
+public:
+  SimpleMDEvent(const std::vector<float> &coordinates, const float &signal, const float &error)
+      : mCoordinates(coordinates), mSignal(signal), mError(error) {}
+
+  SimpleMDEvent(const SimpleMDEvent &other)
+      : mCoordinates(other.mCoordinates), mSignal(other.mSignal), mError(other.mError) {}
+
+  // pretty output
+  std::string str() const {
+    std::stringstream ss;
+    size_t numdirs = mCoordinates.size();
+    for (size_t i = 0; i < numdirs; ++i)
+      ss << mCoordinates[i] << ", ";
+    ss << "signal = " << mSignal << ", error = " << mError;
+
+    return ss.str();
+  }
+
+  float getCenter(size_t dim) const { return mCoordinates[dim]; }
+
+  float getSignal() const { return mSignal; }
+
+  float getError() const { return mError; }
+
+  bool operator()(SimpleMDEvent &event1, const SimpleMDEvent &event2) { return event1 < event2; }
+
+  /**
+   * @brief override operator <
+   * order
+   * 1. coordinate
+   * 2. signal
+   * 3. error
+   * @return
+   */
+  bool operator<(const SimpleMDEvent &event2) const {
+
+    bool less(true);
+    bool equal(false);
+
+    // compare coordinates
+    size_t numdirs = mCoordinates.size();
+    for (size_t i = 0; i < numdirs; ++i) {
+      // equal is hard to tell
+      coord_t diff = mCoordinates[i] - event2.mCoordinates[i];
+
+      if (fabs(mCoordinates[i]) > 1E-7 && fabs(diff / mCoordinates[i]) < 1E-6) {
```

Comment:
Class `SimpleMDEvent` should have a `double static SimpleMDEvent::tolerance` attribute which can be set by algorithm `CompareMDWorkspaces` and default to `1E-07`.

Reply:
float is good enough.  

The `static` is good idea.